src/PackageManagement/AzureAutomationTools.PackageManagement.psm1

```diff
@@ -767,27 +921,60 @@ function DeployModules {
             
         foreach ($Module in $Modules) {
             $ContentLink = $Module.Package
-            $Params = @{Name = $Module.Name; }
+            $Params = @{Name = $Module.Name;}
             $Params += $CommonParameters
-       
-            #FIX failing - version property is coming back blank
-            if ($ExistingModules.Where( {$_.Name -eq $Module.Name -and $_.Version -eq $Module.Version})) {
-                Write-Output "Existing module up to date - skipping."
-            }
-            elseif ($ExistingModules.Where( {$_.Name -eq $Module.Name })) {           
-                Write-Output "Updating existing module ..."
-            
-                Set-AzureRmAutomationModule -ContentLinkUri $ContentLink  @Params
+            $AutomationModule = $null
             
-                Write-Output "Updating existing module - done."
+            try {
+                $ExistingModule = Get-AzureRmAutomationModule @Params
+            }
+            catch [Microsoft.Azure.Commands.Automation.Common.ResourceNotFoundException] {
```

Comment:
The reason I'm using a try/catch here is so that I'm only catching the ResourceNotFoundException. There's no 'friendly' way to ask an Automation Account whether it has a module installed or not. This is a workaround. Would maybe be worth raising with the Azure team.